pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -650,124 +650,125 @@ func TestPullSourcesFromReference(t *testing.T) {
 		SystemRegistriesConfDirPath: "testdata/this-does-not-exist",
 	}
 	registries, err := GetRegistries(sys)
-	assert.Nil(t, err)
-	assert.Equal(t, 2, len(registries))
-
-	// Registry A allowing any kind of pull from mirrors
-	registryA, err := FindRegistry(sys, "registry-a.com/foo/image:latest")
-	assert.Nil(t, err)
-	assert.NotNil(t, registryA)
-	// Digest
-	referenceADigest := toNamedRef(t, "registry-a.com/foo/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
-	pullSources, err := registryA.PullSourcesFromReference(referenceADigest)
-	assert.Nil(t, err)
-	assert.Equal(t, 3, len(pullSources))
-	assert.Equal(t, "mirror-1.registry-a.com/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", pullSources[0].Reference.String())
-	assert.True(t, pullSources[1].Endpoint.Insecure)
-	// Tag
-	referenceATag := toNamedRef(t, "registry-a.com/foo/image:aaa")
-	pullSources, err = registryA.PullSourcesFromReference(referenceATag)
-	assert.Nil(t, err)
-	assert.Equal(t, 3, len(pullSources))
-	assert.Equal(t, "registry-a.com/bar/image:aaa", pullSources[2].Reference.String())
-
-	// Registry B allowing digests pull only from mirrors
-	registryB, err := FindRegistry(sys, "registry-b.com/foo/image:latest")
-	assert.Nil(t, err)
-	assert.NotNil(t, registryB)
-	// Digest
-	referenceBDigest := toNamedRef(t, "registry-b.com/foo/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
-	pullSources, err = registryB.PullSourcesFromReference(referenceBDigest)
-	assert.Nil(t, err)
-	assert.Equal(t, 3, len(pullSources))
-	assert.Equal(t, "registry-b.com/bar", pullSources[2].Endpoint.Location)
-	assert.Equal(t, "registry-b.com/bar/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", pullSources[2].Reference.String())
-	// Tag
-	referenceBTag := toNamedRef(t, "registry-b.com/foo/image:aaa")
-	pullSources, err = registryB.PullSourcesFromReference(referenceBTag)
-	assert.Nil(t, err)
-	assert.Equal(t, 1, len(pullSources))
-}
-
-func TestPullSourcesMirrorFromReference(t *testing.T) {
-	sys := &types.SystemContext{
-		SystemRegistriesConfPath:    "testdata/pull-sources-mirror-reference.conf",
-		SystemRegistriesConfDirPath: "testdata/this-does-not-exist",
-	}
-	registries, err := GetRegistries(sys)
 	require.NoError(t, err)
-	assert.Equal(t, 7, len(registries))
+	assert.Equal(t, 9, len(registries))
 
 	digest := "@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
 	tag := ":aaa"
 	for _, tc := range []struct {
-		registry      string
-		digestSources []string
-		tagSources    []string
+		registry       string
+		repo           string
+		digestSources  []string
```

Comment:
Looking at how these fields (and `tagSources`) are used, they are not really `registry`/`repo`: we have `registry` fields naming namespaces.

So maybe hame the fields (`matchedPrefix`, `repo` (`subRepo`??? probably not), `digestPrefixes`, `tagPrefixes`)?